### PR TITLE
Profiling

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR adds profiling capabilities to theme store.

just change the envvar and start profiling in either dev tools and react profiler

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
[btglobal](https://github.com/vtex-sites/btglobal.store/pull/340)
[marinbrasil](https://github.com/vtex-sites/marinbrasil.store/pull/427)
[storecomponents](https://github.com/vtex-sites/storecomponents.store/pull/691)

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

**Comment by igorbrasileiro** [comment]
As @lbebber said, install the new version of react@experimental and react-dom@experimental solves the problem that profiling and the tree of componente doesn't work. What do you think about bump the version of it?

File: packages/gatsby-theme-store/src/gatsby-node.ts [hunk 1]
```diff
@@ -18,13 +18,15 @@ export interface Options {
...
   defaultLocale: string
+  profiling?: boolean
 }
 
 export const pluginOptionsSchema = ({ Joi }: PluginOptionsSchemaArgs) =>
   Joi.object({
     storeId: Joi.string().required(),
     locales: Joi.array().items(Joi.string()).required(),
     defaultLocale: Joi.string().required(),
+    profile: Joi.boolean(),
```
Comment (klzns) [inline packages/gatsby-theme-store/src/gatsby-node.ts:29]:
```suggestion
    profiling: Joi.boolean(),
```

**Review by klzns — CHANGES_REQUESTED** [review]

File: packages/gatsby-theme-store/src/gatsby-node.ts [hunk 2]
```diff
@@ -178,17 +180,30 @@ export const createPages = async ({
...
 
-export const onCreateWebpackConfig = ({
-  actions: { setWebpackConfig },
-}: CreateWebpackConfigArgs) => {
+export const onCreateWebpackConfig = (
+  { actions: { setWebpackConfig }, stage }: CreateWebpackConfigArgs,
+  { profiling = false }: Options
+) => {
+  const profilingConfig =
+    stage === 'build-javascript' && profiling === true
```
Comment (emersonlaurentino) [inline packages/gatsby-theme-store/src/gatsby-node.ts:188]: this comparison is needed? once profiling will always be boolean

```suggestion
    stage === 'build-javascript' && profiling
```

File: lerna.json [hunk 3]
```diff
@@ -1,13 +1,11 @@
 {
-  "version": "0.286.1",
+  "version": "0.287.1-alpha.1",
```
Comment (emersonlaurentino) [inline lerna.json:2]: why did you generate a version below the existing one?

**Review by emersonlaurentino — APPROVED** [review]

File: packages/gatsby-theme-store/src/gatsby-node.ts (hunk 2 above)
Comment (tlgimenes) [reply packages/gatsby-theme-store/src/gatsby-node.ts:188]: It's always better to explicitly compare values so V8 can better optimize this

**Review by klzns — APPROVED** [review]

**Pull request merged**